Lib/asyncio/base_events.py

```diff
@@ -1531,39 +1533,20 @@ def _run_once(self):
                 handle._run()
         handle = None  # Needed to break cycles when an exception occurs.
 
-    def _set_coroutine_wrapper(self, enabled):
-        try:
-            set_wrapper = sys.set_coroutine_wrapper
-            get_wrapper = sys.get_coroutine_wrapper
-        except AttributeError:
-            return
-
-        enabled = bool(enabled)
-        if self._coroutine_wrapper_set == enabled:
+    def _set_coroutine_origin_tracking(self, enabled):
+        if enabled == self._coroutine_origin_tracking_enabled:
```

Comment:
`if enabled and self._coroutine_origin_tracking_enabled`? I don't like using `==` for bools.

Reply:
No, that would be different – the no-`==` equivalent would be:
```python
if ((enabled and self._coroutine_origin_tracking_enabled) or (not enabled and not self._coroutine_origin_tracking_enabled)):
   ...
```
I find the `==` version easier to read. ("If the requested state equals the current state, return without doing anything.")

Reply:
Got it, then I'd cast them both to `bool`: `if bool(enabled) == bool(self._coroutine_origin_tracking_enabled):`